packages/address/src/utils.ts

```diff
@@ -0,0 +1,87 @@
+import type { BytesLike } from '@ethersproject/bytes';
+import { arrayify, hexlify } from '@ethersproject/bytes';
+import { Logger } from '@ethersproject/logger';
+import { AbstractContract, AbstractWallet } from '@fuel-ts/interfaces';
+import type {
+  Bech32Address,
+  B256Address,
+  AddressLike,
+  ContractIdLike,
+  AbstractAddress,
+} from '@fuel-ts/interfaces';
+import { randomBytes } from '@fuel-ts/keystore';
+import type { Decoded } from 'bech32';
+import { bech32m } from 'bech32';
+
+const logger = new Logger(process.env.BUILD_VERSION || '~');
+
+// Fuel Network HRP (human-readable part) for bech32 encoding
+export const FUEL_BECH32_HRP_PREFIX = 'fuel';
+
+/**
+ * Decodes a Bech32 address string into Decoded
+ */
+export function fromBech32(address: Bech32Address): Decoded {
+  return bech32m.decode(address);
+}
+
+/**
+ * Converts a B256 address string into Bech32
+ */
+export function toBech32(address: B256Address): Bech32Address {
```

Comment:
Why enforce the type B256Address, Bech32Address? I think just string would make easear.

Reply:
Makes sense I think this would help to enforce it: https://github.com/FuelLabs/fuels-ts/pull/441#discussion_r948255727